chat-sdk-ui/src/main/java/co/chatsdk/ui/ImageListAdapter.java

```diff
@@ -0,0 +1,107 @@
+package co.chatsdk.ui;
+
+import android.net.Uri;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+
+public class ImageListAdapter extends RecyclerView.Adapter<ImageListAdapter.MyViewHolder>  {
+
+    public ArrayList<Uri> uriArrayList = new ArrayList<>();
+    private WeakReference<SendMultipleImagesAtOnceActivity> theMainActivity;
+
+    public ImageListAdapter (SendMultipleImagesAtOnceActivity activity) {
+        this.theMainActivity = new WeakReference<SendMultipleImagesAtOnceActivity>(activity);
```

Comment:
Change this to `parentActivity`

Reply:
You don't need to include this bit: 
```
new WeakReference<SendMultipleImagesAtOnceActivity>(activity);
```
only:
```
new WeakReference<>(activity);
```

Reply:
> 
> 
> Change this to `parentActivity`

What should I change to parentActivity? ImageListAdapter?

Reply:
> 
> 
> > Change this to `parentActivity`
> 
> What should I change to parentActivity? ImageListAdapter?

Oh wait, I get it.

Reply:
> 
> 
> You don't need to include this bit:
> 
> ```
> new WeakReference<SendMultipleImagesAtOnceActivity>(activity);
> ```
> 
> only:
> 
> ```
> new WeakReference<>(activity);
> ```

When I do that I get an unchecked assignment message. I hope this is going to be OK.

Reply:
Ok, if you get an error, keep it.